accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,366 @@
+
+Also, SUSE Manager must coordinate the Management Node to issue a correct upgrade procedure of the product (e.g. from CaaS Platform version 3 to 4).
+
+### Application deployment
+
+SUSE Manager can also offer the possibility of deploying an application on top of the cluster deployed (e.g. Kubernetes deployment descriptor to deploy a Kubernetes operator). In this case, the cluster provider must provide the Management Node of the necessary tooling to deploy an application using an API. SUSE Manager makes use of the Management Node to deploy a provided application.
+
+### Specific usage of SUSE Manager features
+
+One cluster deployment can also leverage SUSE Manager features: SUSE CaaS Platform can trigger a CVE Audit on a container image before deploying it. The cluster provider must outline which features intend to use and which interface is to be used (preferred: XMLRPC API).
+
+## Specific section: manage a CaaS Platform cluster using SUSE Manager
+
+### Show cluster status
+
+SUSE Manager shows the status of the cluster by invoking `skuba cluster status` on the Management Node (via Salt).
+
+E.g. of the output:
+
+```
+NAME         OS-IMAGE                              KERNEL-VERSION        CONTAINER-RUNTIME   HAS-UPDATES   HAS-DISRUPTIVE-UPDATES
+cp0       SUSE Linux Enterprise Server 15 SP1   4.12.14-110-default   cri-o://1.13.3      <none> <none>
+worker0   SUSE Linux Enterprise Server 15 SP1   4.12.14-110-default   cri-o://1.13.3      <none> <none>
+```
+
+The output is presented in the "CaaS Platform > cluster name" section.
+
+### Removing nodes from the cluster
+
+A node can also be removed from the cluster:
+
+- temporarily (drained and cordoned off): [using `kubectl`](https://kubernetes.io/docs/tasks/administer-cluster/safely-drain-node/#use-kubectl-drain-to-remove-a-node-from-service) from the Management Node via Salt.
+- permanently: by invoking `skuba node remove <nodename>` via Salt. NOTE: this removal is permanent and the cluster cannot join any cluster in the future (a complete reinstallation is required to so).
+
+Example state:
+
+```
+remove_node:
+      cmd.run:
+        - name: skuba node remove <nodename>
+```
+
+NOTE: Other `skuba` options can be implemented in the future (e.g. resetting and reconfiguring a node).
+
+### Cluster upgrade
+
+All the cluster updates must be handled by `skuba`.
+
+SUSE Manager can check and notify the user if an update for the cluster is present by triggering `skuba cluster upgrade plan` on the Management Node (using Salt) and check if a new output is available in a cron fashion (e.g. Taskomatic job).
+
+SUSE Manager will then interface with the Management node and upgrade every control plane in the cluster first:
+
+`skuba node upgrade apply --target <control-plane-node-ip> --user <user> --sudo`
+
+and then every worker:
+
+`skuba node upgrade apply --target <worker-node-ip> --user <user> --sudo`
+
+This can be done by supplying the command to the Management Node via Salt.
+
```

Comment:
Part of CaaSP's stack is deployed via RPMs, the remaining one is deployed via containers.

`skuba` has a way to upgrade both the pieces of the stack. We will have to update this section once we settle on the "add-ons update story" of CaaSP.

Reply:
Added a note about "add-ons update story future work by CaaSP team"